src/miner.cpp

```diff
@@ -204,8 +204,6 @@ bool BlockAssembler::TestPackage(uint64_t packageSize, int64_t packageSigOpsCost
 
 // Perform transaction-level checks before adding to block:
 // - transaction finality (locktime)
-// - premature witness (in case segwit transactions are added to mempool before
```

Comment:
Should we remove the code with "if (!fIncludeWitness && it->GetTx().HasWitness())"?

Reply:
Perhaps the variable `fIncludeWitness` can be deleted completely?

Reply:
I think everything about witness can be deleted. And in plan to do it, but i already have ~6 PR and i prefer to go step by step :)